src/kudu/util/thread.cc

```diff
@@ -625,4 +616,95 @@ void Thread::FinishThread(void* arg) {
   // following here!
 }
 
+static bool set_capability_flag(cap_value_t capability, cap_flag_value_t flag) {
+  cap_value_t cap_list[]= {capability};
+  cap_t caps= cap_get_proc();
+  bool ret= true;
+
+  if (!caps ||
+      cap_set_flag(caps, CAP_EFFECTIVE, 1, cap_list, flag) ||
+      cap_set_proc(caps)) {
+    LOG(ERROR) << "Can not set capability flag";
+    ret= false;
+  }
+
+  if (caps) {
+    cap_free(caps);
+  }
+
+  return true;
+}
+
+static bool acquire_capability(cap_value_t capability)
+{
+  return set_capability_flag(capability, CAP_SET);
+}
+
+static bool drop_capability(cap_value_t capability)
+{
+  return set_capability_flag(capability, CAP_CLEAR);
+}
+
+// CAP_SYS_NICE capability is acquired before changing the thread priority,
+// and dropped after the action is done.
+// This is the same flow like we did in mysqld
+static int set_system_thread_priority(pid_t tid, int pri)
+{
+  acquire_capability(CAP_SYS_NICE);
+  int ret = setpriority(PRIO_PROCESS, tid, pri) != 0;
+  drop_capability(CAP_SYS_NICE);
+
+  return ret;
+}
+
+static int get_system_thread_priority(pid_t tid)
+{
+  return getpriority(PRIO_PROCESS, tid);
+}
+
+Status ThreadMgr::ShowThreadStatus(vector<ThreadDescriptor>* threads) {
+  for (auto const& name2category : thread_categories_) {
+    ThreadCategory category = name2category.second;
+    for (auto thread_info : category) {
+      int pri = get_system_thread_priority(thread_info.second.thread_id());
+      thread_info.second.setPriority(pri);
+      threads->push_back(thread_info.second);
+    }
+  }
+  return Status::OK();
+}
+
+Status ThreadMgr::ChangeThreadPriority(string category, int priority) {
+  // Change the default for particular pool
+  category2priority_[category] = priority;
```

Comment:
Might need some lock protection. The same for `ThreadMgr::ShowThreadStatus(...)` when accessing thread_categories_.